internal/middleware/etag_middleware.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+)
+
+var InvalidETagErr = errors.New("empty etag provided")
+
+func ETagMiddleware(expires time.Duration, name, etag string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			if etag == "" {
+				panic(InvalidETagErr)
+			}
+			etagValue := fmt.Sprintf("\"hms-%s-%s\"", name, etag)
+			controlValue := fmt.Sprintf("max-age=%d", int(expires.Seconds()))
+			w.Header().Set("ETag", etagValue)
+			w.Header().Set("Cache-Control", controlValue)
+			logger := ctxval.Logger(r.Context()).With().Str("etag", etag).Str("etag_name", name).Logger()
+			logger.Trace().Msgf("Returned etag with Cache-Control '%s'", controlValue)
+
+			if match := r.Header.Get("If-None-Match"); match != "" {
```

Comment:
I've got no idea how this is supposed to be implemented, so it's hard to review this part, why don't we use some external implementation?

Reply:
Yes, I am having the same issue here, can you explain in a few words?

Reply:
While there might be libraries providing such middleware, it is so trivial that I think it is simply worth having it in our codebase.

From Google: https://devcenter.heroku.com/articles/increasing-application-performance-with-http-cache-headers